File: HRMS/HRMS.Web/Modules/Operations/SalaryComponents/SalaryComponentsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace HRMS.Operations.Columns;

[ColumnsScript("Operations.SalaryComponents")]
[BasedOnRow(typeof(SalaryComponentsRow), CheckNames = true)]
public class SalaryComponentsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int ComponentId { get; set; }
    [EditLink, Width(200)]
    public string ComponentName { get; set; }
    [Width(120)]
    public SalaryComponentType ComponentType { get; set; }
    [Width(120)]
    public SalaryCalculationType CalculationType { get; set; }
    [Width(150)]
    public string PercentageOfComponentName { get; set; }
    [Width(80)]
    public bool IsStatutory { get; set; }
    [Width(80)]
    public bool IsTaxable { get; set; }
    [Width(80)]
    public bool IsActive { get; set; }
    [Width(100)]
    public int DisplayOrder { get; set; }
}

File: HRMS/HRMS.Web/Modules/ServerTypes/Operations/SalaryComponentsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { SalaryComponentsRow } from "./SalaryComponentsRow";
import { SalaryComponentType } from "./SalaryComponentType";

export interface SalaryComponentsColumns {
    ComponentId: Column<SalaryComponentsRow>;
    ComponentName: Column<SalaryComponentsRow>;
    ComponentType: Column<SalaryComponentsRow>;
    CalculationType: Column<SalaryComponentsRow>;
    PercentageOfComponentName: Column<SalaryComponentsRow>;
    IsStatutory: Column<SalaryComponentsRow>;
    IsTaxable: Column<SalaryComponentsRow>;
    IsActive: Column<SalaryComponentsRow>;
    DisplayOrder: Column<SalaryComponentsRow>;
}

export class SalaryComponentsColumns extends ColumnsBase<SalaryComponentsRow> {
    static readonly columnsKey = 'Operations.SalaryComponents';
    static readonly Fields = fieldsProxy<SalaryComponentsColumns>();
}

[SalaryComponentType]; // referenced types